app/src/main/java/com/hynekbraun/openmeteoweather/di/RepositoryModule.kt

```diff
@@ -0,0 +1,20 @@
+package com.hynekbraun.openmeteoweather.di
+
+import com.hynekbraun.openmeteoweather.data.repository.WeatherRepositoryImp
+import com.hynekbraun.openmeteoweather.domain.WeatherRepository
+import dagger.Binds
+import dagger.Module
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+abstract class RepositoryModule {
+
+    @Binds
+    @Singleton
```

Comment:
And this goes for all the modules, f.e. for the network module and Retrofit API, it kinda makes sense to have just one instance. But everything that works with it, doesn't have to be a singleton as well.

Reply:
This is a really good info, because to be honest I have been doing it like this automatically, because everybody was doing it and I didn't realize it could be done otherwise. I will definitely read about it.